src/integrated/land.jl

```diff
@@ -0,0 +1,669 @@
+updates the additional auxiliary variables for the integrated model,
+as well as updates the boundary auxiliary variables for all component
+models. 
+
+This function is called each ode function evaluation, prior to the tendency function
+evaluation.
+"""
+function make_update_boundary_fluxes(
+    land::LandModel{FT, MM, SM, RM, SnM},
+) where {
+    FT,
+    MM <: Soil.Biogeochemistry.SoilCO2Model{FT},
+    SM <: Soil.EnergyHydrology{FT},
+    RM <: Canopy.CanopyModel{FT},
+    SnM <: Snow.SnowModel{FT},
+}
+    update_soil_bf! = make_update_boundary_fluxes(land.soil)
+    update_soilco2_bf! = make_update_boundary_fluxes(land.soilco2)
+    update_canopy_bf! = make_update_boundary_fluxes(land.canopy)
+    update_snow_bf! = make_update_boundary_fluxes(land.snow)
+
+    function update_boundary_fluxes!(p, Y, t)
+        earth_param_set = land.soil.parameters.earth_param_set
+        # update root extraction
+        update_root_extraction!(p, Y, t, land) # defined in src/integrated/soil_canopy_root_interactions.jl
+
+        # Radiation - updates Rn for soil and snow also
+        lsm_radiant_energy_fluxes!(
+            p,
+            land,
+            land.canopy.radiative_transfer,
+            Y,
+            t,
+        )
+
+        # Compute the ground heat flux in place:
+        update_soil_snow_ground_heat_flux!(
+            p,
+            Y,
+            land.soil.parameters,
+            land.snow.parameters,
+            land.soil.domain,
+            FT,
+        )
+        #Now update snow boundary conditions, which rely on the ground heat flux
+        update_snow_bf!(p, Y, t)
+
+        # Now we have access to the actual applied and initially computed fluxes for snow
+        @. p.excess_water_flux =
+            (p.snow.total_water_flux - p.snow.applied_water_flux)
+        @. p.excess_heat_flux =
+            (p.snow.total_energy_flux - p.snow.applied_energy_flux)
+
+        # Now we can update the soil BC, and use the excess fluxes there in order to conserv
+        update_soil_bf!(p, Y, t)
+        # Update canopy
+        update_canopy_bf!(p, Y, t)
+        # Update soil CO2
+        update_soilco2_bf!(p, Y, t)
+
+        # compute net flux with atmosphere, this is useful for monitoring conservation
+        _LH_f0 = FT(LP.LH_f0(earth_param_set))
+        _ρ_liq = FT(LP.ρ_cloud_liq(earth_param_set))
+        ρe_falling_snow = -_LH_f0 * _ρ_liq # per unit vol of liquid water
+        @. p.atmos_energy_flux =
+            (1 - p.snow.snow_cover_fraction) * (
+                p.soil.turbulent_fluxes.lhf + p.soil.turbulent_fluxes.shf -
+                p.soil.R_n
+            ) +
+            p.snow.snow_cover_fraction * (
+                p.snow.turbulent_fluxes.lhf + p.snow.turbulent_fluxes.shf -
+                p.snow.R_n
+            ) +
+            p.drivers.P_snow * ρe_falling_snow +
+            p.canopy.energy.turbulent_fluxes.shf +
+            p.canopy.energy.turbulent_fluxes.lhf -
```

Comment:
same sign comment here

Reply:
The signs are changed now so that the only outlier is in soil_canopy_model. This can get fixed in a future PR (will require a change in the conservation runs)